### PR TITLE
OpInfo for `nn.functional.alpha_dropout`

### PR DESCRIPTION
**Pull request by krshrimali**
This PR adds OpInfo for `nn.functional.alpha_dropout`.

Issue tracker: https://github.com/pytorch/pytorch/issues/54261

cc: @mruberry

### CODE REVIEW

**Comment by krshrimali** [comment]
Let's block this PR until https://github.com/pytorch/pytorch/pull/67891 (and if) is merged.

File: test/test_fx_experimental.py [hunk 1]
```diff
@@ -1480,6 +1480,7 @@ def test_normalize_operator_exhaustive(self, device, dtype, op):
             "igammac",
             "index_put",
             "nn.functional.conv2d",
+            "nn.functional.alpha_dropout",
```
Comment (mruberry) [inline test/test_fx_experimental.py:1483]: Can this test be skipped (or better, xfailed) using a DecorateInfo on the OpInfo?

cc @Chillee

**Comment by mruberry** [comment]
This looks good but will need a rebase to get past some test failures -- also see one inline question about the fx test skip

**Comment by krshrimali** [comment]
Waiting for https://github.com/pytorch/pytorch/pull/67891 to land, to be able to reuse sample input of `dropout`.

**Comment by mruberry** [comment]
> Waiting for #67891 to land, to be able to reuse sample input of `dropout`.

Note that #67891 did land

**Comment by krshrimali** [comment]
This should be ready for review now, but I'll ping post the new year (don't wanna disturb during the holiday season) :) Hope you have a happy new year, Ruberry!

File: test/test_fx_experimental.py [hunk 2]
```diff
@@ -1482,12 +1482,9 @@ def test_normalize_operator_exhaustive(self, device, dtype, op):
             "igammac",
             "index_put",
             "nn.functional.conv2d",
-            "nn.functional.dropout",
-            "nn.functional.dropout2d",
             "nn.functional.embedding",  # Implemented with a lambda
             "nn.functional.embedding_bag",  # Implemented with a lambda
             "nn.functional.rrelu",  # Implemented with a lambda
-            "nn.functional.feature_alpha_dropout",  # Implemented with a lambda
```
Comment (krshrimali) [inline test/test_fx_experimental.py:1490]: The dropout variants have been shifted to having an expected failure in their respective OpInfos.

File: test/test_fx_experimental.py (hunk 1 above)
Comment (krshrimali) [reply test/test_fx_experimental.py]: This has been added as an expected failure, also for other dropout variants. Thanks for the suggestion!

**Comment by krshrimali** [comment]
Hi, @mruberry - I hope you're having a good start to the year. In case you are back, could you please take a look at this PR? Thanks!

File: test/test_fx_experimental.py (hunk 2 above)
Comment (mruberry) [reply test/test_fx_experimental.py:1490]: Nice!

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -14440,14 +14446,62 @@ def ref_pairwise_distance(input1, input2):
...
+            # Op implemented with a lambda, see similar issue: https://github.com/pytorch/pytorch/issues/64997
+            DecorateInfo(unittest.expectedFailure, 'TestNormalizeOperators', 'test_normalize_operator_exhaustive'),),
         gradcheck_wrapper=wrapper_set_seed,
         supports_forward_ad=True,
         supports_fwgrad_bwgrad=True,
         supports_out=False,
         sample_inputs_func=partial(sample_inputs_dropout, train=False),
         inplace_variant=lambda input, *args, **kwargs:
             wrapper_set_seed(torch.nn.functional.feature_alpha_dropout, input, *args, **kwargs, inplace=True)),
+    # In training mode, alpha_dropout currently doesn't support inputs of complex dtype
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:14459]: Nice comment

**Comment by mruberry** [comment]
Oh wait -- this was already merged in a different PR, right @krshrimali? Can we close this one?

**Comment by krshrimali** [comment]
Hi, @mruberry - Thank you for taking a look. I checked the master but didn't find the OpInfo for `alpha_dropout`. A [similar PR](https://github.com/pytorch/pytorch/pull/67891) was merged before, but not for `alpha_dropout` (it was `feature_alpha_dropout`) - perhaps you got confused between the two? Please let me know if you meant some other PR, and I can happily close this one if the other one is addressing this.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -14546,9 +14546,11 @@ def ref_pairwise_distance(input1, input2):
             # On CUDA, the op is dispatched (and a few more conditions) to
             # _fused_dropout, which doesn't support forward AD
             DecorateInfo(unittest.skip("Skipped!"), 'TestGradients', 'test_forward_mode_AD', device_type='cuda'),
-            # NotImplementedError: Trying to use forward AD with native_dropout that does not support it
-            DecorateInfo(unittest.expectedFailure, 'TestGradients', 'test_fn_fwgrad_bwgrad',
-                         device_type='cuda', dtypes=[torch.float64]),),
+            # (ROCm) NotImplementedError: Trying to use forward AD with native_dropout that does not support it
```
Comment (IvanYashchuk) [inline torch/testing/_internal/common_methods_invocations.py:14549]: Is `(ROCm)` implies that this test should be skipped only for ROCm and it works on CUDA? 
I don't think we should change `unittest.expectedFailure` -> `unittest.skip("Skipped!")`.

Comment (krshrimali) [reply torch/testing/_internal/common_methods_invocations.py:14549]: Thanks for pointing this out. It was a skip before from [this](https://github.com/pytorch/pytorch/pull/70061/files#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R14561-R14563) PR and was changed to `expectedFailure` [here](https://github.com/pytorch/pytorch/pull/71944). I think I missed something during the rebase. I'll fix this now.

**Comment by krshrimali** [comment]
Gentle ping, @mruberry! PTAL, whenever you find the time.

**Comment by krshrimali** [comment]
Gentle ping, @saketh-are @mruberry

**Pull request closed without merging**